realtimenet/finetuning.py

```diff
@@ -196,4 +211,41 @@ def run_epoch(data_loader, net, criterion, optimizer=None, use_gpu=False):
     top1 = np.mean(epoch_labels == epoch_top_predictions)
     loss = running_loss / len(data_loader)
 
-    return loss, top1
+    cnf_matrix = confusion_matrix(epoch_labels, epoch_top_predictions)
+
+    return loss, top1, cnf_matrix
+
+
+def plot_confusion_matrix(path_out, cm, classes,
+                          normalize=False, title='Confusion matrix', cmap=plt.cm.Blues):
+    """
+    This function prints and plots the confusion matrix.
+    Normalization can be applied by setting `normalize=True`.
+    """
+    plt.figure()
+    plt.imshow(cm, interpolation='nearest', cmap=cmap)
+    plt.title(title)
+    plt.colorbar()
+    tick_marks = np.arange(len(classes))
+    plt.xticks(tick_marks, classes, rotation=90)
+    plt.yticks(tick_marks, classes)
+
+    if normalize:
+        cm = cm.astype('float') / cm.sum(axis=1)[:, np.newaxis]
+        print("Normalized confusion matrix")
+    else:
+        print('Confusion matrix, without normalization')
+
+    thresh = cm.max() / 2.
+    for i, j in itertools.product(range(cm.shape[0]), range(cm.shape[1])):
+        plt.text(j, i, cm[i, j],
+                 horizontalalignment="center",
+                 color="white" if cm[i, j] > thresh else "black")
+
+    # plt.tight_layout()
+    plt.ylabel('True label')
+    plt.xlabel('Predicted label')
+    plt.savefig(os.path.join(path_out, 'confusion_matrix.png'), bbox_inches='tight')
```

Comment:
plt.savefig(os.path.join(path_out, 'confusion_matrix.png'), transparent=False, pad_inches=0.1, bbox_inches='tight', dpi = 300)

The dpi might improve the quality of the image

Reply:
Done. Worked well and fixed the misalignment issue